roborock/device_trait.py

```diff
@@ -0,0 +1,57 @@
+from abc import ABC, abstractmethod
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+
+from . import RoborockCommand
+from .containers import DeviceFeatures, RoborockBase
+
+
+@dataclass
+class DeviceTrait(ABC):
+    handle_command: RoborockCommand
+
+    def __init__(self, send_command: Callable[..., Awaitable[None]]):
+        self.send_command = send_command
+        self.status: RoborockBase | None = None
+        self.subscriptions = []
+
+    @classmethod
+    @abstractmethod
+    def supported(cls, features: DeviceFeatures) -> bool:
+        raise NotImplementedError
+
+    @abstractmethod
+    def from_dict(cls, data: dict) -> bool:
+        raise NotImplementedError
+
+    def on_message(self, data: dict) -> None:
+        self.status = self.from_dict(data)
```

Comment:
Typically `from_dict` is a factor method so here i think its used more like `update`.